src/ui/ui_settings/ui_settings_service.js

```diff
@@ -8,6 +8,17 @@ function hydrateUserSettings(userSettings) {
     .reduce((acc, { key, userValue }) => ({ ...acc, [key]: { userValue } }), {});
 }
 
+/**
+ * Invalid value for UiSetting
+ * @class InvalidValudError
+ */
+export class InvalidValueError extends Error {
```

Comment:
We have a fairly established convention for handling service-specific errors. See: 

 -
 [SavedObjectsClient/errors](https://github.com/elastic/kibana/blob/131b63c0f8b294083f77a141016e053c2a19d811/src/server/saved_objects/client/lib/errors.js)
 - [SavedObjectsClient#errors](https://github.com/elastic/kibana/blob/131b63c0f8b294083f77a141016e053c2a19d811/src/server/saved_objects/client/saved_objects_client.js#L94-L95)

Reply:
Is this an approach that we intend to support with the new platform? Throwing boom errors, which are http specific seems like it won't scale that well as we start to support more "background" tasks that aren't attached to a specific http request/response.